### ISSUE TITLE
UpdateEntitlements returns past already consumed purchases

### ISSUE THREAD

**Issue by WiseKodama** [qonversion/unity-sdk#246]
In our app we offer only consumable purchases. 

At the start of the app once the purchases/products are loaded we call `Qonversion.GetSharedInstance().CheckEntitlements()` since this is what is in the documentation:
```You need to call the checkEntitlements() method at the start of your app to check if a user has the required entitlement. This method will check the user's receipt and return current entitlements.```
This returns already consumed purchases and their entitlements. 

Note that I did not verify the boolean value of `IsActive` of the returned entitlement.

As far as I know consumable purchase support is still in progress, correct?

@SpertsyanKM tagging you here since you were able to help me before, apologies if this is someone elses territory.

**Comment by SpertsyanKM** [comment]
Hello, @WiseKodama. 

If you are selling only consumable products, what entitlements do you check in your app? Entitlements are intended for subscriptions and non-consumable in-apps. Can you explain your use-case please?

**Comment by WiseKodama** [comment]
Hello @SpertsyanKM, great to hear from you again :) 

Basically entitlements are cross linked with our products in the dashboard. Basically for easier identification after a purchase has completed.

We also needed to use `qonversionInstance.UpdatedEntitlementsReceived += OnEntitlementsUpdated;` in order to handle pending purchases. 

As for the issue at hand we are unsure why it would return already consumed entitlements?
It shows that there is support for consumable purchases: https://hub.qonversion.io/roadmap/consumable-in-apps-support-from-the-sdk

Basically our use cases are:
* Handle purchase
* Handle pending purchase(app in background)
* Handle pending purchase(app killed) -> Purchase should be rewarded upon start if purchase has completed outside app

**Comment by SpertsyanKM** [comment]
Sorry for the late reply.

While we support consumable in-app purchases, we don't support pending purchases for that type of products. This means you can't identify such a purchase with an updated entitlements listener, cause it doesn't actually grant an entitlement (or the entitlement should also be consumable, but that is not so for now).

In a couple of days, we'll discuss with the team the priority of this feature. As soon as I have any news, I'll share them here.

**Comment by WiseKodama** [comment]
I already tested pending purchases and the cases mentioned worked.

However recently it started occurring that the SDK was returning already
consumed entitlements if the app was reinstalled.

So something must’ve changed to cause this issue.
Please let me know if there are any workarounds for the issue at hand.

**Comment by SpertsyanKM** [comment]
If you link your in-app product to some entitlement, we will grant a lifetime entitlement for the purchase of that product (no matter it is consumable or not) - that is the expected behavior for now. That is why you receive an entitlement for the consumed purchase

**Comment by WiseKodama** [comment]
Apologies for the slow response @SpertsyanKM 

Are there any plans to make Qonversion work with Consumable purchases that will also cover pending/out-of-app purchases?

**Comment by SpertsyanKM** [comment]
Yes, we plan to implement this feature by the end of this year.

**Comment by WiseKodama** [comment]
Hey @SpertsyanKM 👋 
Any chance there is an issue/task tracker for non-subscription purchases support?

**Comment by SpertsyanKM** [comment]
Hi @WiseKodama,

I think, you can follow this [feature request](https://hub.qonversion.io/b/zv9ko901/feature-ideas/consumable-entitlements) in our hub.

**Comment by WiseKodama** [comment]
Hey @SpertsyanKM 

I see that the feature is still marked as `Under consideration`. 
Will it make it to release this year?

**Comment by WiseKodama** [comment]
Hey @SpertsyanKM 

Any news with this [feature request](https://hub.qonversion.io/b/zv9ko901/feature-ideas/consumable-entitlements)?

How much effort would it be to send data if the purchase that completed was in a pending state beforehand?
Is that something I can do on my end? If so, could you point me in the correct direction please?

**Comment by SpertsyanKM** [comment]
We do not offer purchase as an entity on the SDK side, so there is no information returned for pending purchases. Only when the purchase is successful will the updated entitlements listener be triggered.
